sql-plugin/src/main/332/scala/com/nvidia/spark/rapids/shims/SparkShims.scala

```diff
@@ -16,4 +16,21 @@
 
 package com.nvidia.spark.rapids.shims
 
-object SparkShimImpl extends Spark331PlusShims with AnsiCastRuleShims
+import com.nvidia.spark.rapids._
+
+import org.apache.spark.sql.execution.command.{CreateDataSourceTableAsSelectCommand, DataWritingCommand, RunnableCommand}
+
+object SparkShimImpl extends Spark331PlusShims with AnsiCastRuleShims {
+  override def getDataWriteCmds: Map[Class[_ <: DataWritingCommand],
```

Comment:
Duplicated block. Extract this block to `311until340-all`

Reply:
Do you mean adding a new trait `Spark31Xuntil34X`?

Reply:
I mean `311until340-all`, this PR is handing a new Spark change from version 340.
So all the previous versions [311, 340) are able to use this code block, we can extract this to a trait to reuse it.